### DISCUSSION TITLE
Options for 2D Vector Graphics Drawing

### DISCUSSION THREAD

**Discussion by mdmozibur** [unoplatform/uno#18028] [Q&A]
I want to build an app for Mac Catalyst from an existing UWP app. Researching the internet, I understand that Uno Platform is the best options for me.
The basic need for the app is to be able to draw 2D geometries. In the UWP app, this was handled by Win2D very comfortably. And the reason for choosing Win2D over something like SkiaSharp was mainly performance(hardware acceleration) issues and rich features of Win2D.
Upon researching, I came to understand that the direct alternative for Win2D in MAC is [CoreGrahics](https://developer.apple.com/documentation/coregraphics). My question is, Is it possible for my UWP app to be converted to a MAC app with the requirements mentioned above? What is the state of 2D Graphics functionality in Uno? I have found Uno.SkiaSharp to be the closest option available. Does it translate to CoreGraphics commands? Will hardware acceleration be available in MAC if I choose Uno.SkiaSharp?
What should be my best course of action here?

**Comment by jeromelaban** [comment]
Thanks for the question.

This particular topic is being actively worked on at this time, particularly in https://github.com/unoplatform/uno/pull/16621, which will provide a hardware accelerated canvas support. The new `SKCanvas` will transparently use the underlying hardware acceleration, OpenGL or Metal, depending on the platform.

Still, as of Uno Platform 5.3, the current option is to use `SkiaSharp` through `SKXamlCanvas`, which only supports software rendering.

**Comment by serefarikan** [comment]
@jeromelaban just to understand what Uno can do: is SKCanvas available when using the wasm deployment/environment?

**Reply by jeromelaban** [reply]
The upcoming `SKCanvasElement` is only available on Skia rendered targets for desktop at this time.

**Reply by TopperDEL** [reply]
As I just stumbled across this: will there be a hardware accelerated canvas for WASM? Or is this not possible due to $InsertTechnicalRestrictionHere?

**Reply by jeromelaban** [reply]
`SKSwapChainPanel` is hardware accelerated on Wasm.

**Reply by TopperDEL** [reply]
Oh! So I just have to use that one instead? Nice! Will try.
